azkaban-common/src/main/java/azkaban/jmx/JmxExecutionController.java

```diff
@@ -33,7 +33,7 @@ public JmxExecutionController(final ExecutionController controller) {
 
   @Override
   public int getNumRunningFlows() {
-    return this.controller.getRunningFlows().size();
+    return this.controller.getRunningFlowIds().size();
```

Comment:
I searched this `getRunningFlowIds().size()` it appears 4 times in this single PR. Also we know getRunningFlows this is on our hotpath. 
So I think it's worth to have a separate method that we count from DB directly.

Reply:
Agree but I think it is fine. 3 out of 4 such places occur in JMX* but we didn't enable JMX metrics server internally.